tests/test_utility.py

```diff
@@ -23,55 +28,39 @@ def test_add_trailing_slash(url, expected_url):
-            {"http://firstnode.neurobagel.org/query/": "firstnode"},
-        ),
-        (
-            "(https://firstnode.neurobagel.org/query/, firstnode) (https://secondnode.neurobagel.org/query, secondnode)",
-            {
-                "https://firstnode.neurobagel.org/query/": "firstnode",
-                "https://secondnode.neurobagel.org/query/": "secondnode",
-            },
-        ),
-        (
-            "(firstnode.neurobagel.org/query/, firstnode) (https://secondnode.neurobagel.org/query, secondnode)",
-            {
-                "https://secondnode.neurobagel.org/query/": "secondnode",
-            },
-        ),
-        (
-            "( , firstnode) (https://secondnode.neurobagel.org/query, secondnode)",
-            {
-                "https://secondnode.neurobagel.org/query/": "secondnode",
-            },
-        ),
-        (
-            "(https://firstnode.neurobagel.org/query/, firstnode)(https://secondnode.neurobagel.org/query, secondnode)",
             {
-                "https://firstnode.neurobagel.org/query/": "firstnode",
-                "https://secondnode.neurobagel.org/query/": "secondnode",
+                "ApiURL": "http://firstnode.neurobagel.org/query",
+                "NodeName": "firstnode",
             },
+            {"http://firstnode.neurobagel.org/query/": "firstnode"},
         ),
         (
-            "(https://firstnode.neurobagel.org/query/,firstnode)(https://secondnode.neurobagel.org/query,secondnode)",
+            [
+                {
+                    "ApiURL": "https://firstnode.neurobagel.org/query/",
+                    "NodeName": "firstnode",
+                },
+                {
+                    "ApiURL": "https://secondnode.neurobagel.org/query",
+                    "NodeName": "secondnode",
+                },
+            ],
             {
                 "https://firstnode.neurobagel.org/query/": "firstnode",
                 "https://secondnode.neurobagel.org/query/": "secondnode",
             },
         ),
+        ({}, {}),
     ],
 )
-def test_parse_nodes_as_dict(monkeypatch, set_nodes, expected_nodes):
-    """Test that Neurobagel node URLs provided in a string environment variable are correctly parsed into a list."""
-    monkeypatch.setenv("LOCAL_NB_NODES", set_nodes)
-    # TODO: This currently only compares the keys of the dicts, not the values, due to calling sorted(). This is probably not what we want.
-    assert sorted(
-        util.parse_nodes_as_dict(
-            os.environ.get(
-                "LOCAL_NB_NODES", "https://api.neurobagel.org/query/"
-            )
-        )
-    ) == sorted(expected_nodes)
+def test_parse_nodes_as_dict(
+    set_nodes, expected_nodes, tmp_local_nb_nodes_path
+):
+    """Test that Neurobagel nodes provided via a JSON file are correctly parsed into a list."""
+    with open(tmp_local_nb_nodes_path, "w") as f:
+        f.write(json.dumps(set_nodes, indent=2))
```

Comment:
I'd give this a little comment as a reminder that we first create the physical input file for the test function to read here